cmd/metal-api/internal/service/service.go

```diff
@@ -33,6 +33,50 @@ type webResource struct {
 	ds  *datastore.RethinkStore
 }
 
+// logger returns the request logger from the request.
+func (w *webResource) logger(rq *restful.Request) *zap.SugaredLogger {
+	return rest.GetLoggerFromContext(rq.Request, w.log)
+}
+
+func (w *webResource) sendError(rq *restful.Request, rsp *restful.Response, httperr *httperrors.HTTPErrorResponse) {
+	w.logger(rq).Errorw("service error", "status", httperr.StatusCode, "error", httperr.Message, "service-caller", utils.CallerFuncName(2))
```

Comment:
Can be probably replaced with: https://pkg.go.dev/go.uber.org/zap?utm_source=godoc#AddCaller